tests/step-119.sh

```diff
@@ -0,0 +1,115 @@
+#!/bin/sh
+# Test:
+#   Test legacy transform introduced in PR #125 by updating from 49464f09f0 to HEAD
+#   - split global githooks.shared values : `legacy_transform_adjust_local_paths`
+#   - move local paths in .shared to local githooks.shared : `legacy_transform_split_global_shared_entries`
+
+cleanup() {
+    # Remove the whole install to
+    # not make external script which calls uninstall
+    # to dispatch into this uninstall!
+    git hooks uninstall --global >/dev/null 2>&1
+}
+
+trap 'cleanup' EXIT
+
+git clone https://github.com/rycus86/githooks.git /var/lib/githooks-original
+
+cd /var/lib/githooks-original &&
+    git checkout master &&
+    git checkout -b master-old &&
+    git checkout master &&
+    git reset --hard 49464f09f0 || exit 1
+
+# Install old version
+if ! sh /var/lib/githooks/install.sh --clone-url /var/lib/githooks-original --clone-branch "master"; then
+    echo "! Failed to execute the install script"
+    exit 1
+fi
+
+# Make shared hook repo
+makeShared() {
+    idx="$1"
+    mkdir -p "/tmp/shared$idx.git" &&
+        cd "/tmp/shared$idx.git" &&
+        git init &&
+        mkdir -p .githooks/pre-commit &&
+        git config core.hooksPath "/dev/null" && # dont execute hooks in this repo!
+        echo "echo 'Shared repo $idx: pre-commit'" >.githooks/pre-commit/sample-one &&
+        git add . &&
+        git commit -a -m 'Testing 1' ||
+        return 1
+}
+
+makeShared 1 || exit 1
+makeShared 2 || exit 2
+makeShared 3 || exit 3
+makeShared 4 || exit 4
+
+# Install global hooks
+git config --global githooks.shared "file:///tmp/shared1.git,/tmp/shared2.git"
+
+# Make repo with local paths in .shared file
+mkdir -p /tmp/test119 && cd /tmp/test119 || exit 7
+git init &&
+    mkdir ".githooks" &&
+    touch ".githooks/trust-all" &&
+    echo "# Hooks A" >>.githooks/.shared &&
+    echo "file:///tmp/shared3.git" >>.githooks/.shared &&
+    echo "# Hooks B" >>.githooks/.shared &&
+    echo "/tmp/shared4.git" >>.githooks/.shared &&
+    git add . || exit 8
+
+git hooks config accept trusted || exit 9
+git hooks shared update || exit 10
+
+OUT=$(git commit -a -m "Testing" 2>&1)
+if ! echo "$OUT" | grep -q "Shared repo 1: pre-commit" ||
+    ! echo "$OUT" | grep -q "Shared repo 2: pre-commit" ||
+    ! echo "$OUT" | grep -q "Shared repo 3: pre-commit" ||
+    ! echo "$OUT" | grep -q "Shared repo 4: pre-commit"; then
+    echo "! Expected to have run 4 shared hooks: $OUT" >&2
+    exit 10
+fi
+
+# Update to new version
+git config --global githooks.testingTreatFileProtocolAsRemote "true"
+
+cd /var/lib/githooks-original &&
+    git reset --hard master-old &&
+    cp -r /var/lib/githooks/* ./ &&
+    git commit -a -m "Current Changes: Simulating a merge with this feature branch" || exit 11
```

Comment:
I *think* you could also do something like `cd /var/lib/githooks-original && git fetch /var/lib/githooks && git reset --hard FETCH_HEAD` here, to let git handle everything.

Alternatively, when cloning into `githooks-original` above, you could maybe just clone from `/var/lib/githooks` rather than than github (also makes the test more self-contained), and then you can be sure that `/var/lib/githook`'s current branch is already available in `githooks-original` (as `master-old`) so you can just skip to there?

Reply:
I cant clone from `/var/lib/githooks` -> it contains at least 50 commits from travis (if we would do this) currently its a self build small repo with two commtis done by the Dockerfile

Reply:
> `git fetch /var/lib/githooks && git reset --hard FETCH_HEAD`
Hows that gona work? :-) Dont get it. So you mean we basically pull in changes from another repo. 
`/var/lib/githooks` contains some initial commit + the current version currently... (its not the history as github)

Reply:
Ah, I was assuming that `/var/lib/githooks` was just a clone of this repo with the commit to be tested checked out, but apparently things are a lot more complicated than that. It smells like things could maybe simplified here, but I'm not going to dive into this part too :-p

Reply:
You are right! It could in my opinion be simplified. Maybe another PR!